cpp/ql/src/semmle/code/cpp/Macro.qll

```diff
@@ -275,6 +277,8 @@ deprecated class MacroInvocationExpr extends Expr {
     exists(MacroInvocation i | this = i.getExpr())
   }
 
+  override string getCanonicalQLClass() { result = "MacroInvocationExpr" }
```

Comment:
`MacroInvocationExpr` and `MacroInvocationStmt` are deprecated subclasses.  Make `MacroAccess` and/or `MacroInvocation` canonical instead.

Reply:
I just tried what you suggested, but do not get any mentions of `MacroAccess` or `MacroInvocation`. It would appear that `MacroInvocationExpr` and `MacroInvocationStmt` are still being used in spite of being `deprecated`, and so I think I'll leave them in for now.  We can change this in the future, possibly once our test cases get broken.

Reply:
I strongly disagree.  `MacroInvocationExpr` and `MacroInvocationStmt` are deprecated for a reason (people tend to misunderstand what they do and do not match, and write queries that don't quite behave as they expect).  I'm not sure what you mean about them still being used.  We should not point users to them through `getCanonicalQLClass`.

I think `MacroAccess` should have a `getCanonicalQLClass`.  There are [lots of them](https://lgtm.com/query/2563812634614495030/?pageIndex=0), though our older queries more commonly interact with them through the `MacroInvocation` subclass.